esp/bindings/SOAP/Platform/soapservice.cpp

```diff
@@ -185,7 +185,8 @@ int CSoapService::processRequest(ISoapMessage &req, ISoapMessage& resp)
     EndTag etag;
     StringBuffer peerStr;
     ctx->getPeer(peerStr);
-    const char* userId = ctx->queryUserId();
+    StringBuffer uid(ctx->queryUserId());
+    const char* userId = uid.str();
```

Comment:
@RussWhitehead looks like the userID is changed after the processHeader() call. You may need to retrieve the userID after that call to make sure that userID is correct.